src/Categories/Bicategory/Monad/Bimodule.agda

```diff
@@ -0,0 +1,99 @@
+{-# OPTIONS --without-K --safe --lossy-unification #-}
+
+open import Categories.Bicategory
+
+module Categories.Bicategory.Monad.Bimodule {o ℓ e t} {𝒞 : Bicategory o ℓ e t} where
+
+open import Level
+open import Categories.Bicategory.Monad using (Monad)
+import Categories.Bicategory.Extras as Bicat
+open Bicat 𝒞
+
+record Bimodule (M₁ M₂ : Monad 𝒞) : Set (o ⊔ ℓ ⊔ e) where
+  open Monad using (C; T; μ; η)
+  field
+    F       : C M₁ ⇒₁ C M₂
+    actionˡ : F ∘₁ T M₁ ⇒₂ F
+    actionʳ : T M₂ ∘₁ F ⇒₂ F
+
+    assoc     : actionʳ ∘ᵥ (T M₂ ▷ actionˡ) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionʳ ◁ T M₁)
+    sym-assoc : actionˡ ∘ᵥ (actionʳ ◁ T M₁) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T M₂ ▷ actionˡ)
+
+    assoc-actionˡ     : actionˡ ∘ᵥ (F ▷ μ M₁) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionˡ ◁ T M₁)
+    sym-assoc-actionˡ : actionˡ ∘ᵥ (actionˡ ◁ T M₁) ∘ᵥ associator.to ≈ actionˡ ∘ᵥ (F ▷ μ M₁)
+    assoc-actionʳ     : actionʳ ∘ᵥ (μ M₂ ◁ F) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T M₂ ▷ actionʳ)
+    sym-assoc-actionʳ : actionʳ ∘ᵥ (T M₂ ▷ actionʳ) ∘ᵥ associator.from ≈ actionʳ ∘ᵥ (μ M₂ ◁ F)
+
+    identityˡ : actionˡ ∘ᵥ (F ▷ η M₁) ∘ᵥ unitorʳ.to ≈ id₂
+    identityʳ : actionʳ ∘ᵥ (η M₂ ◁ F) ∘ᵥ unitorˡ.to ≈ id₂
+
+id-bimodule : (M : Monad 𝒞) → Bimodule M M
+id-bimodule M = record
+  { F = T
+  ; actionˡ = μ
+  ; actionʳ = μ
+  ; assoc = assoc
+  ; sym-assoc = sym-assoc
+  ; assoc-actionˡ = assoc
+  ; sym-assoc-actionˡ = sym-assoc
+  ; assoc-actionʳ = sym-assoc
+  ; sym-assoc-actionʳ = assoc
+  ; identityˡ = identityˡ
+  ; identityʳ = identityʳ
+  }
+  where
+    open Monad M
+
+record Bimodulehomomorphism {M₁ M₂ : Monad 𝒞} (B₁ B₂ : Bimodule M₁ M₂) : Set (ℓ ⊔ e) where
```

Comment:
I know that for smaller things, the objects and morphisms are all defined in the same file, but I think that for `Bimodule`, it would make sense to split this into `Bimodule/Homomorphism.agda`. It could also help make typechecking faster.